python/src/spark_rapids_ml/knn.py

```diff
@@ -514,24 +455,10 @@ async def do_allGather() -> List[str]:
 
         return _cuml_fit
 
-    def _transform(self, dataset: DataFrame) -> DataFrame:
-        raise NotImplementedError(
-            "NearestNeighborsModel does not provide a transform function. Use 'kneighbors' instead."
-        )
-
-    def _get_cuml_transform_func(
-        self, dataset: DataFrame
-    ) -> Tuple[
-        Callable[..., CumlT],
-        Callable[[CumlT, Union[cudf.DataFrame, np.ndarray]], pd.DataFrame],
-    ]:
-        raise NotImplementedError(
-            "'_CumlModel._get_cuml_transform_func' method is not implemented. Use 'kneighbors' instead."
-        )
-
     def exactNearestNeighborsJoin(
         self,
         query_df: DataFrame,
+        item_df: DataFrame,
         distCol: str = "distCol",
```

Comment:
One open question is which of query_df and item_df comes first in the argument list. 

Spark LSH has an API `approxNearestNeighbors(dataset, key, ...)` that puts item_df before the query vector.